### ISSUE TITLE
求助，路径正确，但总是无法导入 yolov3_coco_demo.ckpt，无法加载预训练模型

### ISSUE THREAD

**Issue by Byronnar** [YunYang1994/tensorflow-yolov3#297]
因为我改动了.names文件，类别不一样了，想问一下，如果不使用原版coco.names就不能加载预训练模型吗？也就是说训练自己的数据必须从头开始？那还有什么意义呢？
希望有大神可以指导一下

**Comment by YunYang1994** [comment]
检查下你的tf版本

**Comment by Byronnar** [comment]
> 检查下你的tf版本

哇，作者回复了，请问检查哪个版本？tensorflow的嘛？

**Comment by Byronnar** [comment]
> 检查下你的tf版本

我是1.8的版本

**Comment by Byronnar** [comment]
> 检查下你的tf版本

我把自己的names文件名字改成coco.names就出现了train loss nan的情况，一直是这样，yolov3_pretrained.ckpt是yolov3预训练模型转过来的
=> Restoring weights from: ./checkpoint/yolov3_pretrained.ckpt ... 
train loss: 286.56: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5811/5811 [20:13<00:00,  4.29it/s]
=> Epoch:  1 Time: 2019-08-26 10:47:19 Train loss: 418.41 Test loss: 255.67 Saving ./checkpoint/yolov3_train_loss_418.4113.ckpt ...
train loss: nan: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5811/5811 [40:29<00:00,  2.91it/s]
=> Epoch:  2 Time: 2019-08-26 11:28:02 Train loss: nan Test loss: nan Saving ./checkpoint/yolov3_train_loss_nan.ckpt ...

**Comment by mengqp1994** [comment]
> 因为我改动了.names文件，类别不一样了，想问一下，如果不使用原版coco.names就不能加载预训练模型吗？也就是说训练自己的数据必须从头开始？那还有什么意义呢？
> 希望有大神可以指导一下



> 因为我改动了.names文件，类别不一样了，想问一下，如果不使用原版coco.names就不能加载预训练模型吗？也就是说训练自己的数据必须从头开始？那还有什么意义呢？
> 希望有大神可以

> 因为我改动了.names文件，类别不一样了，想问一下，如果不使用原版coco.names就不能加载预训练模型吗？也就是说训练自己的数据必须从头开始？那还有什么意义呢？
> 希望有大神可以指导一下

将train.py的108行 self.loader = tf.train.Saver(self.net_var)
改为
variables_to_restore = [v for v in self.net_var if v.name.split('/')[0] not in ['conv_sbbox','conv_mbbox','conv_lbbox']]
self.loader = tf.train.Saver(variables_to_restore)

**Comment by Byronnar** [comment]
> > 因为我改动了.names文件，类别不一样了，想问一下，如果不使用原版coco.names就不能加载预训练模型吗？也就是说训练自己的数据必须从头开始？那还有什么意义呢？
> > 希望有大神可以指导一下
> 
> > 因为我改动了.names文件，类别不一样了，想问一下，如果不使用原版coco.names就不能加载预训练模型吗？也就是说训练自己的数据必须从头开始？那还有什么意义呢？
> > 希望有大神可以
> 
> > 因为我改动了.names文件，类别不一样了，想问一下，如果不使用原版coco.names就不能加载预训练模型吗？也就是说训练自己的数据必须从头开始？那还有什么意义呢？
> > 希望有大神可以指导一下
> 
> 将train.py的108行 self.loader = tf.train.Saver(self.net_var)
> 改为
> variables_to_restore = [v for v in self.net_var if v.name.split('/')[0] not in ['conv_sbbox','conv_mbbox','conv_lbbox']]
> self.loader = tf.train.Saver(variables_to_restore)

真的非常感谢你，可以进行预训练了。赞

**Comment by jersonal** [comment]
> > 因为我改动了.names文件，类别不一样了，想问一下，如果不使用原版coco.names就不能加载预训练模型吗？也就是说训练自己的数据必须从头开始？那还有什么意义呢？
> > 希望有大神可以指导一下
> 
> > 因为我改动了.names文件，类别不一样了，想问一下，如果不使用原版coco.names就不能加载预训练模型吗？也就是说训练自己的数据必须从头开始？那还有什么意义呢？
> > 希望有大神可以
> 
> > 因为我改动了.names文件，类别不一样了，想问一下，如果不使用原版coco.names就不能加载预训练模型吗？也就是说训练自己的数据必须从头开始？那还有什么意义呢？
> > 希望有大神可以指导一下
> 
> 将train.py的108行 self.loader = tf.train.Saver(self.net_var)
> 改为
> variables_to_restore = [v for v in self.net_var if v.name.split('/')[0] not in ['conv_sbbox','conv_mbbox','conv_lbbox']]
> self.loader = tf.train.Saver(variables_to_restore)

这样改动了，如果中断训练后想要继续训练，会出现loss重新开始
